cmd/nerdctl/pull.go

```diff
@@ -120,73 +120,27 @@ func pullAction(cmd *cobra.Command, args []string) error {
 }
 
 func ensureImage(ctx context.Context, cmd *cobra.Command, client *containerd.Client, globalOptions types.GlobalCommandOptions, rawRef string, ocispecPlatforms []v1.Platform, pull string, unpack *bool, quiet bool) (*imgutil.EnsuredImage, error) {
-
 	var ensured *imgutil.EnsuredImage
 
 	verifier, err := cmd.Flags().GetString("verify")
 	if err != nil {
-		return nil, err
+		return ensured, err
 	}
-
-	if scheme, ref, err := referenceutil.ParseIPFSRefWithScheme(rawRef); err == nil {
-		if verifier != "none" {
-			return nil, errors.New("--verify flag is not supported on IPFS as of now")
-		}
-
-		ipfsAddressStr, err := cmd.Flags().GetString("ipfs-address")
-		if err != nil {
-			return nil, err
-		}
-
-		var ipfsPath *string
-		if ipfsAddressStr != "" {
-			dir, err := os.MkdirTemp("", "apidirtmp")
-			if err != nil {
-				return nil, err
-			}
-			defer os.RemoveAll(dir)
-			if err := os.WriteFile(filepath.Join(dir, "api"), []byte(ipfsAddressStr), 0600); err != nil {
-				return nil, err
-			}
-			ipfsPath = &dir
-		}
-
-		ensured, err = ipfs.EnsureImage(ctx, client, cmd.OutOrStdout(), cmd.ErrOrStderr(), globalOptions.Snapshotter, scheme, ref,
-			pull, ocispecPlatforms, unpack, quiet, ipfsPath)
-		if err != nil {
-			return nil, err
-		}
-		return ensured, nil
+	ipfsAddressStr, err := cmd.Flags().GetString("ipfs-address")
+	if err != nil {
+		return ensured, err
 	}
 
-	ref := rawRef
-	switch verifier {
-	case "cosign":
-		experimental := globalOptions.Experimental
-
-		if !experimental {
-			return nil, fmt.Errorf("cosign only work with enable experimental feature")
-		}
-
-		keyRef, err := cmd.Flags().GetString("cosign-key")
-		if err != nil {
-			return nil, err
-		}
-
-		ref, err = cosignutil.VerifyCosign(ctx, rawRef, keyRef, globalOptions.HostsDir)
-		if err != nil {
-			return nil, err
-		}
-	case "none":
-		logrus.Debugf("verification process skipped")
-	default:
-		return nil, fmt.Errorf("no verifier found: %s", verifier)
+	options := types.PullCommandOptions{
+		GOptions:    globalOptions,
+		Verify:      verifier,
+		IPFSAddress: ipfsAddressStr,
```

Comment:
All command options construct processes should be in processPullCommandFlags

Reply:
Done.